iphone/Maps/Core/Location/MWMLocationManager.mm

```diff
@@ -528,8 +527,6 @@ - (void)stop
 - (void)updateFrameworkInfo
 {
   auto app = UIApplication.sharedApplication;
-  if (app.applicationState != UIApplicationStateActive)
-    return;
   auto delegate = static_cast<MapsAppDelegate *>(app.delegate);
   if (delegate.isDrapeEngineCreated)
```

Comment:
The problem with this approach is that it calls itself without any delay, while in reality GPS locations come every second. @fabwu did you measure how many times is this method called before actually updating a location? A hundred times or more?

I suggest to properly move this code into https://github.com/organicmaps/organicmaps/blob/master/iphone/Maps/Core/Location/MWMLocationManager.mm#L305

Reply:
> @fabwu did you measure how many times is this method called before actually updating a location? A hundred times or more?

I put a log statement and it looks like it doesn't recurse.

If you want a proper refactoring, I would move the framework update to a separate observer and register that observer with `[MWMLocationManager addObserver]`. Then the framework gets location updates like the rest of the app (e.g. `MWMRouter`). We could do the same for the GPS tracker.

Reply:
@fabwu you're right, framework update is called only once, it recurses only if drape engine wasn't created yet for any reason (that should never happen).

What are the pros and cons of using observer pattern instead of a simpler direct call of `f.OnLocationUpdate` and friends where necessary?

I see in the profiler on the simulator that most of the CPU in the background is spent in onNavigationInfoUpdated. As the dashboard is not visible, it shouldn't be updated every time in the background, right? Maybe make a separate issue about it?